File: 06-dynamic/ReflectAndroid/app/src/main/java/com/ztiany/reflect/AppContext.java
package com.ztiany.reflect;

import android.app.Application;
import android.content.Context;
import android.util.Log;

/**
 * @author Ztiany
 *         Email: ztiany3@gmail.com
 *         Date : 2017-12-28 23:31
 */
public class AppContext extends Application {

    private static final String TAG = AppContext.class.getSimpleName();

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        ActivityThreadHook.getInstance().hookActivityThread();
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "onCreate() called");
    }
}

File: 05-architecture/Kotlin-Android-V1.2/app/src/main/java/com/ztiany/kotlin/AppContext.kt
package com.ztiany.kotlin

import android.app.Application
import android.content.Context

/**
 *@author Ztiany
 *      Email: ztiany3@gmail.com
 *      Date : 2017-07-09 18:51
 */
class AppContext : Application() {

    private object C {
        lateinit var instance: AppContext
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        C.instance = this
    }

    companion object {
        fun getInstance(): AppContext {
            return C.instance
        }
    }


}